src/Database/Database.php

```
@@ -4452,7 +4453,11 @@ public function updateDocuments(
             foreach ($batch as $doc) {
                 $this->purgeCachedDocument($collection->getId(), $doc->getId());
                 $doc = $this->decode($collection, $doc);
-                $onNext && $onNext($doc);
+                try {
+                    $onNext && $onNext($doc);
+                } catch (Exception $e) {
+                    $onError && $onError($e);
```

I think we need to throw the exception in case no onerror callback?

---

According to the linear ticket, the execution should go and if error comes it should get passed to the onError callback
If rethrown the execution will stop